### PR TITLE
Specifies build args for frontend

### PR DESCRIPTION
**Pull request by snuderl**
Also specifies dockerfile for custom builds, to let skaffold auto infer dependencies.

### CODE REVIEW

File: skaffold.yaml [hunk 1]
```diff
@@ -48,3 +60,22 @@ profiles:
...
+          session_api_client_base_url: https://session-api.dev.snuderls.eu
+          auth_api_client_base_url: https://auth-api.dev.snuderls.eu
+          auth_api_server_base_url: http://auth-api
+          try_app_base_url: https://try.dev.snuderls.eu
+          bootstrap_script: https://static.dev.snuderls.eu/b/development.insight.js
+          app_base_url: https://app.dev.snuderls.eu
+      - op: add
+        path: /build/artifacts/1/docker/buildArgs
+        value:
+          session_api_client_base_url: https://session-api.dev.snuderls.eu
```
Comment (Meemaw) [inline skaffold.yaml:76]: Not needed

File: skaffold.yaml [hunk 2]
```diff
@@ -48,3 +60,22 @@ profiles:
...
+          auth_api_client_base_url: https://auth-api.dev.snuderls.eu
+          auth_api_server_base_url: http://auth-api
```
Comment (Meemaw) [inline skaffold.yaml:78]: Not needed

File: skaffold.yaml [hunk 3]
```diff
@@ -48,3 +60,22 @@ profiles:
...
+          try_app_base_url: https://try.dev.snuderls.eu
```
Comment (Meemaw) [inline skaffold.yaml:79]: Not needed

File: skaffold.yaml [hunk 4]
```diff
@@ -48,3 +60,22 @@ profiles:
...
+    patches:
+      - op: add
+        path: /build/artifacts/0/docker/buildArgs
+        value:
+          session_api_client_base_url: https://session-api.dev.snuderls.eu
+          auth_api_client_base_url: https://auth-api.dev.snuderls.eu
+          auth_api_server_base_url: http://auth-api
+          try_app_base_url: https://try.dev.snuderls.eu
+          bootstrap_script: https://static.dev.snuderls.eu/b/development.insight.js
+          app_base_url: https://app.dev.snuderls.eu
```
Comment (Meemaw) [inline skaffold.yaml:72]: Not needed

File: skaffold.yaml [hunk 5]
```diff
@@ -48,3 +60,22 @@ profiles:
     build:
       local:
         push: true
+    patches:
+      - op: add
+        path: /build/artifacts/0/docker/buildArgs
+        value:
+          session_api_client_base_url: https://session-api.dev.snuderls.eu
```
Comment (Meemaw) [inline skaffold.yaml:67]: Add `session_api_server_base_url`: http://session-api

File: skaffold.yaml (hunk 4 above)
Comment (snuderl) [reply skaffold.yaml]: its simpler to just do the same for both :)

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

**Plan Error**
```
running "/usr/local/bin/terraform plan -input=false -refresh -no-color -out \"/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github/default.tfplan\"" in "/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github": exit status 1
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


Error: If `anonymous` is false, `token` is required.

  on providers.tf line 1, in provider "github":
   1: provider "github" {



```

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

**Plan Error**
```
running "/usr/local/bin/terraform plan -input=false -refresh -no-color -out \"/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github/default.tfplan\"" in "/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github": exit status 1
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


Error: If `anonymous` is false, `token` is required.

  on providers.tf line 1, in provider "github":
   1: provider "github" {



```

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

**Plan Error**
```
running "/usr/local/bin/terraform plan -input=false -refresh -no-color -out \"/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github/default.tfplan\"" in "/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github": exit status 1
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


Error: If `anonymous` is false, `token` is required.

  on providers.tf line 1, in provider "github":
   1: provider "github" {



```

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

**Plan Error**
```
running "/usr/local/bin/terraform plan -input=false -refresh -no-color -out \"/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github/default.tfplan\"" in "/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github": exit status 1
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


Error: If `anonymous` is false, `token` is required.

  on providers.tf line 1, in provider "github":
   1: provider "github" {



```

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

<details><summary>Show Output</summary>

```diff

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
+ create
~ update in-place

Terraform will perform the following actions:

  # github_repository.insight will be updated in-place
~ resource "github_repository" "insight" {
        allow_merge_commit     = false
        allow_rebase_merge     = false
        allow_squash_merge     = true
        archived               = false
        default_branch         = "master"
        delete_branch_on_merge = true
        description            = "Find insights into your frontend applications."
        etag                   = "W/\"2c56e82c74bea706238c703a9b91ced1\""
        full_name              = "insights-io/Insight"
        git_clone_url          = "git://github.com/insights-io/Insight.git"
        has_downloads          = false
        has_issues             = true
        has_projects           = true
        has_wiki               = true
        html_url               = "https://github.com/insights-io/Insight"
        http_clone_url         = "https://github.com/insights-io/Insight.git"
        id                     = "Insight"
        is_template            = false
        name                   = "Insight"
        node_id                = "MDEwOlJlcG9zaXRvcnkyNzA5NTAzMDU="
        private                = false
        ssh_clone_url          = "git@github.com:insights-io/Insight.git"
        svn_url                = "https://github.com/insights-io/Insight"
      ~ topics                 = [
          - "elasticsearch",
          - "github-actions",
          - "gradle",
          - "jaegertracing",
          - "java",
          - "javascript",
          - "kafka",
          - "kubernetes",
          - "postgresql",
          - "quarkus",
          - "react",
          - "storybook",
          - "typescript",
          - "vertx",
        ]
    }

  # github_repository.ops will be created
+ resource "github_repository" "ops" {
      + allow_merge_commit     = false
      + allow_rebase_merge     = false
      + allow_squash_merge     = false
      + archived               = false
      + default_branch         = "master"
      + delete_branch_on_merge = false
      + description            = "Insight operations"
      + etag                   = (known after apply)
      + full_name              = (known after apply)
      + git_clone_url          = (known after apply)
      + has_issues             = false
      + has_projects           = false
      + has_wiki               = false
      + html_url               = (known after apply)
      + http_clone_url         = (known after apply)
      + id                     = (known after apply)
      + name                   = "ops"
      + node_id                = (known after apply)
      + private                = false
      + ssh_clone_url          = (known after apply)
      + svn_url                = (known after apply)
    }

  # module.branch_protection.github_branch_protection.branch_protection will be updated in-place
~ resource "github_branch_protection" "branch_protection" {
        branch                 = "master"
      ~ enforce_admins         = false -> true
        etag                   = "W/\"7b5570235231cf93a47f04b1d120f4a0\""
        id                     = "Insight:master"
        repository             = "Insight"
        require_signed_commits = false

        required_pull_request_reviews {
            dismiss_stale_reviews           = true
            dismissal_teams                 = []
            dismissal_users                 = []
            include_admins                  = false
            require_code_owner_reviews      = false
            required_approving_review_count = 1
        }

        required_status_checks {
            contexts       = []
            include_admins = false
            strict         = true
        }
    }

Plan: 1 to add, 2 to change, 0 to destroy.

```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -d infrastructure/terraform/github`
* :put_litter_in_its_place: To **delete** this plan click [here](https://atlantis.dev.snuderls.eu/lock?id=insights-io%252FInsight%252Finfrastructure%252Fterraform%252Fgithub%252Fdefault)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -d infrastructure/terraform/github`
</details>

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

<details><summary>Show Output</summary>

```diff

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
+ create
~ update in-place

Terraform will perform the following actions:

  # github_repository.insight will be updated in-place
~ resource "github_repository" "insight" {
        allow_merge_commit     = false
        allow_rebase_merge     = false
        allow_squash_merge     = true
        archived               = false
        default_branch         = "master"
        delete_branch_on_merge = true
        description            = "Find insights into your frontend applications."
        etag                   = "W/\"e6b98d4da6e0481104f6d5966cbfcf4a\""
        full_name              = "insights-io/Insight"
        git_clone_url          = "git://github.com/insights-io/Insight.git"
        has_downloads          = false
        has_issues             = true
        has_projects           = true
        has_wiki               = true
        html_url               = "https://github.com/insights-io/Insight"
        http_clone_url         = "https://github.com/insights-io/Insight.git"
        id                     = "Insight"
        is_template            = false
        name                   = "Insight"
        node_id                = "MDEwOlJlcG9zaXRvcnkyNzA5NTAzMDU="
        private                = false
        ssh_clone_url          = "git@github.com:insights-io/Insight.git"
        svn_url                = "https://github.com/insights-io/Insight"
      ~ topics                 = [
          - "elasticsearch",
          - "github-actions",
          - "gradle",
          - "jaegertracing",
          - "java",
          - "javascript",
          - "kafka",
          - "kubernetes",
          - "postgresql",
          - "quarkus",
          - "react",
          - "storybook",
          - "typescript",
          - "vertx",
        ]
    }

  # github_repository.ops will be created
+ resource "github_repository" "ops" {
      + allow_merge_commit     = false
      + allow_rebase_merge     = false
      + allow_squash_merge     = false
      + archived               = false
      + default_branch         = "master"
      + delete_branch_on_merge = false
      + description            = "Insight operations"
      + etag                   = (known after apply)
      + full_name              = (known after apply)
      + git_clone_url          = (known after apply)
      + has_issues             = false
      + has_projects           = false
      + has_wiki               = false
      + html_url               = (known after apply)
      + http_clone_url         = (known after apply)
      + id                     = (known after apply)
      + name                   = "ops"
      + node_id                = (known after apply)
      + private                = true
      + ssh_clone_url          = (known after apply)
      + svn_url                = (known after apply)
    }

  # module.branch_protection.github_branch_protection.branch_protection will be updated in-place
~ resource "github_branch_protection" "branch_protection" {
        branch                 = "master"
      ~ enforce_admins         = false -> true
        etag                   = "W/\"7b5570235231cf93a47f04b1d120f4a0\""
        id                     = "Insight:master"
        repository             = "Insight"
        require_signed_commits = false

        required_pull_request_reviews {
            dismiss_stale_reviews           = true
            dismissal_teams                 = []
            dismissal_users                 = []
            include_admins                  = false
            require_code_owner_reviews      = false
            required_approving_review_count = 1
        }

        required_status_checks {
            contexts       = []
            include_admins = false
            strict         = true
        }
    }

Plan: 1 to add, 2 to change, 0 to destroy.

```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -d infrastructure/terraform/github`
* :put_litter_in_its_place: To **delete** this plan click [here](https://atlantis.dev.snuderls.eu/lock?id=insights-io%252FInsight%252Finfrastructure%252Fterraform%252Fgithub%252Fdefault)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -d infrastructure/terraform/github`
</details>

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`

**Comment by Meemaw** [comment]
atlantis apply

**Comment by botsights** [comment]
Ran Apply for dir: `infrastructure/terraform/github` workspace: `default`

**Apply Failed**: Pull request must be approved by at least one person other than the author before running apply.

**Comment by Meemaw** [comment]
atlantis apply

**Comment by botsights** [comment]
Ran Apply for dir: `infrastructure/terraform/github` workspace: `default`

**Apply Error**
<details><summary>Show Output</summary>

```
running "/usr/local/bin/terraform apply -input=false -no-color \"/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github/default.tfplan\"" in "/atlantis-data/repos/insights-io/Insight/240/default/infrastructure/terraform/github": exit status 1
github_repository.ops: Creating...
github_repository.insight: Modifying... [id=Insight]
github_repository.insight: Modifications complete after 6s [id=Insight]
module.branch_protection.github_branch_protection.branch_protection: Modifying... [id=Insight:master]
module.branch_protection.github_branch_protection.branch_protection: Modifications complete after 3s [id=Insight:master]

Error: POST https://api.github.com/orgs/insights-io/repos: 422 Repository creation failed. [{Resource:Repository Field: Code:custom Message:Either :allow_merge_commit, :allow_squash_merge or :allow_rebase_merge must be true.}]

  on repositories.tf line 23, in resource "github_repository" "ops":
  23: resource "github_repository" "ops" {



```
</details>

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

<details><summary>Show Output</summary>

```diff

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
+ create

Terraform will perform the following actions:

  # github_repository.ops will be created
+ resource "github_repository" "ops" {
      + allow_merge_commit     = false
      + allow_rebase_merge     = false
      + allow_squash_merge     = true
      + archived               = false
      + default_branch         = "master"
      + delete_branch_on_merge = false
      + description            = "Insight operations"
      + etag                   = (known after apply)
      + full_name              = (known after apply)
      + git_clone_url          = (known after apply)
      + has_issues             = false
      + has_projects           = false
      + has_wiki               = false
      + html_url               = (known after apply)
      + http_clone_url         = (known after apply)
      + id                     = (known after apply)
      + name                   = "ops"
      + node_id                = (known after apply)
      + private                = true
      + ssh_clone_url          = (known after apply)
      + svn_url                = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -d infrastructure/terraform/github`
* :put_litter_in_its_place: To **delete** this plan click [here](https://atlantis.dev.snuderls.eu/lock?id=insights-io%252FInsight%252Finfrastructure%252Fterraform%252Fgithub%252Fdefault)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -d infrastructure/terraform/github`
</details>

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`

**Comment by Meemaw** [comment]
atlantis apply

**Comment by botsights** [comment]
Ran Apply for dir: `infrastructure/terraform/github` workspace: `default`

```diff
github_repository.ops: Creating...
github_repository.ops: Creation complete after 4s [id=ops]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

```

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

```diff

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.

```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -d infrastructure/terraform/github`
* :put_litter_in_its_place: To **delete** this plan click [here](https://atlantis.dev.snuderls.eu/lock?id=insights-io%252FInsight%252Finfrastructure%252Fterraform%252Fgithub%252Fdefault)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -d infrastructure/terraform/github`

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

```diff

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.

```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -d infrastructure/terraform/github`
* :put_litter_in_its_place: To **delete** this plan click [here](https://atlantis.dev.snuderls.eu/lock?id=insights-io%252FInsight%252Finfrastructure%252Fterraform%252Fgithub%252Fdefault)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -d infrastructure/terraform/github`

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`

**Comment by botsights** [comment]
Ran Plan for dir: `infrastructure/terraform/github` workspace: `default`

```diff

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.

```

* :arrow_forward: To **apply** this plan, comment:
    * `atlantis apply -d infrastructure/terraform/github`
* :put_litter_in_its_place: To **delete** this plan click [here](https://atlantis.dev.snuderls.eu/lock?id=insights-io%252FInsight%252Finfrastructure%252Fterraform%252Fgithub%252Fdefault)
* :repeat: To **plan** this project again, comment:
    * `atlantis plan -d infrastructure/terraform/github`

---
* :fast_forward: To **apply** all unapplied plans from this pull request, comment:
    * `atlantis apply`

**Comment by botsights** [comment]
Locks and plans deleted for the projects and workspaces modified in this pull request:

- dir: `infrastructure/terraform/github` workspace: `default`

**Pull request merged**